### PR TITLE
[11.0][ADD] pos_order_mgmt: New module

### PR DESCRIPTION
**Pull request by chienandalu**
This module extends the functionality of the PoS frontend allowing to load
already done PoS Orders in order to be able to operate over them, being able to
reprint past tickets o return them.

Based on @legalsylvain work here https://github.com/grap/grap-odoo-incubator/tree/8.0/pos_done_order_load but reconducted to manage the loaded orders from a list screen and filling the shortcomings of that one.

Hopefully it can be useful as a base for other modules.

cc @Tecnativa

### CODE REVIEW

File: pos_order_list/models/pos_config.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from odoo import fields, models
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    iface_load_done_order = fields.Boolean(
+        string='Load Done Orders',
+        default=False,
```
Comment (legalsylvain) [inline pos_order_list/models/pos_config.py]:
```suggestion
        default=True,
```

We can guess that if users install the module, he want to see the feature enabled.

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @chienandalu. Thanks for porting this module !
Looks nice. (one comment inline about default value.)
other things : 
- could you take too from the original module, the screen shot in static / description and reference in the readme part (or add new one) ? It's good in OCA appstore to see what is the screen provided by the module.
- I have an error, testing on runbot, when I try to reprint or reload the pos.order. did you faced the same problem ?

kind regards.

image

**Comment by chienandalu** [comment]
Mm, I'm testing but I couldn't reproduce the error either from new or reloaded orders. What steps did you follow?

**Comment by chienandalu** [comment]
Ok, I tested with Firefox and failed as you described. It should be fixed right now.

File: pos_order_list/README.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/184/11.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module extends the functionality of the PoS frontend allowing to load
+already done PoS Orders in order to be able to operate over them, being able to
+reprint past tickets o return them.
```
Comment (legalsylvain) [inline pos_order_list/README.rst]:
```suggestion
reprint past tickets or return them.
```

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi. Thanks !
Well, testing on rubot, Seems that this module has feature already placed in another OCA module 'pos_order_return'. 
So actually, you can see that if you return an order on runbot, with the other module, an other error occures when validating.

image


solution I see : 
- merge the two modules. (big work) 
- make your depend on the other one, and send the value of the refunded order. returned_order_id
https://github.com/OCA/pos/blob/11.0/pos_order_return/models/pos_order.py#L14
(more easy, just overload the JS function that prepare the order.

what do you think ?

kind regards.

**Comment by chienandalu** [comment]
@legalsylvain It was kinda tricky (we're dealing with synced and unsynced orders) but I could make both modules get along together :slightly_smiling_face:

**Comment by pedrobaeza** [comment]
@legalsylvain is everything correct now?

**Comment by chienandalu** [comment]
@legalsylvain Could you re-review again?

**Comment by rafaelbn** [comment]
Hello @legalsylvain ! Could you please RE-Review this PR? We are running it in production and I think that it could be very useful to have it merged for more visibility about OCA POS improvements in App Store. As much users they use as more mature will be. It looks like other Payment modules are more popular.

Thanks

**Comment by pedrobaeza** [comment]
hahaha, we (the 2) have requested at the same time the same thing.

**Comment by pedrobaeza** [comment]
(this is the result of an internal chat conversation)

**Comment by legalsylvain** [comment]
Hi @chienandalu. First, thanks for the changes.
All is working like a charm. (tested on runbot) reprint, refund.
the only missing feature that in pos_order_return you can not refund many times the same order. (more precisely, for each line, you can not refund more than the original quantity).
this is handled with that part of code in ``pos.order.line`` : https://github.com/OCA/pos/blob/11.0/pos_order_return/models/pos_order.py#L135

## Use case
**via back office (pos_order_return)** 
- select an order A, make a refund, confirm.
- select the order A again, make a refund : the refund is empty. OK

**Front office (pos_order_mngmt)**
- select an order A, make a refund, confirm.
- select the order A again, make a refund  You can refund many times. KO

This is visible with the partial refund wizard :
image

**solution 1** :  you introduce returned_line_id in your Javasript code (can be a big work, to handle all the cases)

**solution 2** : you introduce a roadmap / know issue in your rst file, mentionning that the check in front office is not implemented. (less work ! ;-)) 

kind regards.

otherwise, LGTM !

**Comment by chienandalu** [comment]
> he only missing feature that in pos_order_return you can not refund many times the same order

@legalsylvain Ok, I'll let it for a future revision as we invested a lot of time in this already :wink:

**Review by legalsylvain — APPROVED** [review]
OK. I permit me to update the roadmap file in your branch to mention that this feature is partially implemented.

let's go merging this PR.

thanks for this work !

**Comment by pedrobaeza** [comment]
@chienandalu Travis is red

**Comment by chienandalu** [comment]
@pedrobaeza Yes, there's an issue with pos_loyalty yet to be resolved: https://github.com/OCA/pos/issues/309

**Comment by pedrobaeza** [comment]
OK, then let's merge it right now

**Pull request merged**